libs/core/src/components/pds-select/test/pds-select.spec.tsx

```diff
@@ -0,0 +1,229 @@
+import { newSpecPage } from '@stencil/core/testing';
+import { PdsSelect } from '../pds-select';
+
+import { enlarge } from '@pine-ds/icons/icons';
+
+describe('pds-select', () => {
+  it('renders', async () => {
+    const { root } = await newSpecPage({
+      components: [PdsSelect],
+      html: `<pds-select component-id="field-1"></pds-select>`,
+    });
+    expect(root).toEqualHtml(`
+      <pds-select component-id="field-1">
+        <mock:shadow-root>
+          <div class="pds-select">
+            <label htmlFor="field-1"></label>
+            <select class="pds-select__field" id="field-1">
+            </select>
+            <div aria-hidden="true" class="hidden">
+              <slot></slot>
+            </div>
+            <pds-icon class="pds-select__select-icon" icon="${enlarge}"></pds-icon>
+          </div>
+        </mock:shadow-root>
+      </pds-select>
+    `);
+  });
+
+  it('renders with slotted options', async () => {
+    const page = await newSpecPage({
+      components: [PdsSelect],
+      html: `<pds-select component-id="field-1" label="Name">
+               <option value="1">Option 1</option>
+               <option value="2">Option 2</option>
+             </pds-select>`,
+    });
+
+    // Trigger the handleSlotChange logic
+    const slot = page.root?.shadowRoot?.querySelector('slot');
+    const select = page.root?.shadowRoot?.querySelector('select');
+    const slottedOptions = slot?.assignedNodes?.({ flatten: true });
+
+    slottedOptions?.forEach((node) => {
+      if (node.nodeType === Node.ELEMENT_NODE) {
+        select?.appendChild(node.cloneNode(true));
+      }
+    });
```

Comment:
Do you need this here? Is there another way that the internal function `handleSlotChange` could be called?

Reply:
This is from your refactor to fix a failing spec. I can def look into this once all the fixes/updates are applied.